src/openrct2/platform/shared.c

```diff
@@ -509,10 +510,28 @@ void platform_process_messages()
 				int gesturePixels = (int)(_gestureRadius * gScreenWidth);
 				if (gesturePixels > tolerance) {
 					_gestureRadius = 0;
-					keyboard_shortcut_handle_command(SHORTCUT_ZOOM_VIEW_IN);
+					
+					if (gScreenFlags & SCREEN_FLAGS_TITLE_DEMO)
+						break;
+					if (!(gScreenFlags & SCREEN_FLAGS_SCENARIO_EDITOR) || gS6Info.editor_step == EDITOR_STEP_LANDSCAPE_EDITOR) {
```

Comment:
This part (and the similar part) is hard to read and contains duplication. It's probably better to put them into a temporary bool and read that.